Fix deserialization of maps of bing tiles on the client

mbasmanova opened the pull request:
In most cases objects of BingTileType are serialized as JSON. However, when used as keys in a map they are serialized as strings. Hence, the client needs to handle both cases. 

Here is what the client receives in response to `select map(array[bing_tile(1, 2, 10)], array[bing_tile(3, 4, 11)]);`

`"data":[[{"BingTile{x=1, y=2, zoom_level=10}":{"x":3,"y":4,"zoom":11}}]]`

Fixes #9693.

```
presto> select bing_tile(1,2,3);
select bing_tile(1,2,3);
       _col0        
--------------------
 {x=1, y=2, zoom=3} 
(1 row)

presto> select array[bing_tile(1, 2, 10)];
select array[bing_tile(1, 2, 10)];
         _col0         
-----------------------
 [{x=1, y=2, zoom=10}] 
(1 row)

presto> select map(array[bing_tile(1, 2, 10)], array[bing_tile(3, 4, 11)]);
select map(array[bing_tile(1, 2, 10)], array[bing_tile(3, 4, 11)]);
                          _col0                          
---------------------------------------------------------
 {BingTile{x=1, y=2, zoom_level=10}={x=3, y=4, zoom=11}} 
(1 row)
```

nezihyigitbasi commented on presto-client/src/main/java/com/facebook/presto/client/QueryResults.java:
```diff
@@ -332,6 +333,10 @@ private static Object fixValue(TypeSignature signature, Object value)
             case CHAR:
             case GEOMETRY:
                 return String.class.cast(value);
+            case BING_TILE:
```
nit: 
```
// Bing tiles are serialized as strings when used as map keys/values,
// they are serialized as json otherwise (value will be a LinkedHashMap).
```

mbasmanova commented:
@nezihyigitbasi , thank you for review. I added a test, updated the commit message and a comment in the code as you suggested. The build is failing, but looks like it might be a transient failure. Is there a way to trigger and re-run of the tests?

nezihyigitbasi commented:
I restarted the tests.

nezihyigitbasi commented:
Merged, thanks!
